### ISSUE TITLE
resource sources should receive start/end params

### ISSUE THREAD

**Issue by arshaw** [fullcalendar/fullcalendar-scheduler#246]
The [`resources` function](https://fullcalendar.io/docs/resource_data/resources_function/) should now begin to receive a `start`, `end`, and `timezone` parameter just like [`events` function](https://fullcalendar.io/docs/event_data/events_function/).

The [`resources` feed](https://fullcalendar.io/docs/resource_data/resources_json_feed/) should probably also receive these params for the network request, but only if `refetchResourcesOnNavigate` is enabled.

**Comment by arshaw** [comment]
Workaround for the `resources` function:

```js
resources: function(callback, view) {
    var view = $('#calendar').fullCalendar('getView');
    /* use view.start and view.end, which are both moments */
    /* call callback, just like you normally would */
}
```

Workaround for `resources` as an json feed (relies on a resources function):

```js
resources: function(callback, view) {
    var view = $('#calendar').fullCalendar('getView');
    $.ajax({
        url: 'your feed url',
        dataType: 'json',
        cache: false
    }).then(resources) {
        callback(resources);
    });
}
```

**Comment by kpollardsprypoint** [comment]
Thank you @arshaw !

**Comment by gcheema** [comment]
+1

**Comment by CraigHoneyfieldIS** [comment]
I noticed when I try to get the start date as mentioned above for resources, the date returned by 
view.start.format(); = the prior date that the user was viewing, it's not the new date that is clicked on to navigate to.  

I'm using the j-query calendar as a navigator and setting the date via 
`$('#calendar').fullCalendar('gotoDate', d);`
Any suggestions?

**Comment by Spartaseventy** [comment]
I have the same problem of @CraigHoneyfieldIS.
I need to refetch resources  (and events) on day change and view change, but i'm encountering various problem.
This is my script code
`$(document).ready(function(){
		var $calendar = $("#calendar");
		$calendar.fullCalendar({
			schedulerLicenseKey: "GPL-My-Project-Is-Open-Source",
			editable: true,
			allDaySlot: false,
			aspectRatio: 2,
			scrollTime: "00:00",
			minTime: "00:00:00",
			slotDuration: "00:15:00",
			slotLabelFormat: "H:mm",
			slotWidth: "10%",
			timeFormat: "H:mm",
			header: {
				left: "today prev,next",
				center: "title",
				right: "timelineDay,agendaWeek,month,listWeek"
			},
			defaultView: "timelineDay",
			refetchResourcesOnNavigate: true,
			resourceLabelText: "Siti",
 			resourceGroupField: "site",
			resourceAreaWidth: "25%",
			resources: function(callback) {
			    var view = $("#calendar").fullCalendar("getView");
			    $.ajax({
					url: "myurl",
					dataType: "json",
					cache: false,
					data: {
						start: view.start.format(),
						end: view.end.format(),
						timezone: view.options.timezone
					}
				}).then(function(resources) {
					callback(resources);
				})
			}
		});
	})`
but on load i have this error in console `view.start is null`
If i add a condition to check if value of start and end of view is null bypassing error i show the resources but when i change day or view the date returned by view.start or view.end or view.intervalStart or view.intervalEnd are the prior date that the user was viewing, it's not the new date that is clicked on to navigate to.
It's a bug or i make same mistakes?

I take this opportunity to wish you happy new year

Thanks

**Comment by arshaw** [comment]
@CraigHoneyfieldIS thanks, I understand the issue. I'll make a real fix for this soon, but a good ol' `setTimeout` will actually solve the problem. JSBin proof: http://jsbin.com/hiwidavixo/1/edit?js,output.

I have updated my previously posted sample code to include the timeout hack.

**Comment by margusarm** [comment]
Unfortunately this doesn't work for me, there is something messed up in the end of the code? too many/few parentheses or something else missing. error: `Uncaught SyntaxError: Unexpected token {` and it is pointing to the line `}).then(resources) {`


```
resources: function(callback) {
    setTimeout(function() { // necessary hack
        var view = $('#calendar').fullCalendar('getView'); // TODO: update to your selector
        $.ajax({
            url: 'your feed url',
            dataType: 'json',
            cache: false,
            data: {
                start: view.start.format(),
                end: view.end.format(),
                timezone: view.options.timezone
            }
        }).then(resources) {
            callback(resources);
        });
    }, 0);
}

```

**Comment by arshaw** [comment]
Released in [v1.5.1](https://github.com/fullcalendar/fullcalendar-scheduler/releases/tag/v1.5.1)

updated docs:
https://fullcalendar.io/docs/resource_data/refetchResourcesOnNavigate/
https://fullcalendar.io/docs/resource_data/resources_function/
https://fullcalendar.io/docs/resource_data/resources_json_feed/
